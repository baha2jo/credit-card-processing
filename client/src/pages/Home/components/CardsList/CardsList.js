import React, { useEffect } from 'react';
import { useSelector } from 'react-redux';
import { CardItem } from '../CardItem';
import { fetchCards } from 'actions/card.action';
import useAction from 'hooks/useAction';

export const CardsList = () => {
  const [fetchCardsAction] = useAction(fetchCards);
  const cards = useSelector(state => state.cards);

  useEffect(() => {
    fetchCardsAction();
  }, [fetchCardsAction]);

  return (
    <div className='mt-4 h-100 overflow-auto'>
      <h4 className='text-center font-weight-lighter'>Cards List</h4>
      <div className='line'></div>
      <div className='h-100 overflow-auto'>
        {cards.map(item => (
          <div className='card-item' key={item._id}>
            <CardItem card={item} />
          </div>
        ))}
        {cards.length === 0 && (
          <h3 className='text-center mt-4 font-weight-lighter'>
            No Cards Available
          </h3>
        )}
      </div>
    </div>
  );
};
