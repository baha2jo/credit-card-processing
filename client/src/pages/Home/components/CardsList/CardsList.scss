.cards-list {
  max-height: 685px;
}
