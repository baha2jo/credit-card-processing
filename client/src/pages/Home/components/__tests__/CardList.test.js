import { CardsList } from '../CardsList';

function setup() {
  const store = global.mockStore({
    cards: [
      {
        cardNumber: '4970856264139166',
        expiryDate: '12/21',
        limit: '122',
        name: '3213',
        _id: '2QCfp3SjZ7bMo1gi'
      }
    ]
  });

  return { wrapper: global.mockRender(store, <CardsList />) };
}

describe('CardsList component', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
