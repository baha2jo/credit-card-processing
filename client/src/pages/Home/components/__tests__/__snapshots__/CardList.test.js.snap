// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsList component should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <CardsList>
    <div
      className="mt-4 h-100 overflow-auto"
    >
      <h4
        className="text-center font-weight-lighter"
      >
        Cards List
      </h4>
      <div
        className="line"
      />
      <div
        className="h-100 overflow-auto"
      >
        <CardItem
          card={
            Object {
              "_id": "2QCfp3SjZ7bMo1gi",
              "cardNumber": "4970856264139166",
              "expiryDate": "12/21",
              "limit": "122",
              "name": "3213",
            }
          }
          key="2QCfp3SjZ7bMo1gi"
        >
          <div
            className="card-item"
          >
            <div
              className="d-flex align-items-center"
            >
              <span
                className="card-item__icon mr-4"
              >
                <CardIcon
                  number="4970856264139166"
                >
                  <ForwardRef(SvgVisa)>
                    <svg>
                      visa.svg
                    </svg>
                  </ForwardRef(SvgVisa)>
                </CardIcon>
              </span>
              <div>
                <p
                  className="mb-0"
                >
                  4970856264139166
                </p>
                <p
                  className="mb-0"
                >
                  3213
                </p>
                <small
                  className="text-muted"
                >
                  12/21
                </small>
              </div>
            </div>
            <p
              className="text-right text-info"
            >
              122
              $
            </p>
            <div
              className="line"
            />
          </div>
        </CardItem>
      </div>
    </div>
  </CardsList>
</Provider>
`;
